src/handler/mod.rs

```diff
@@ -552,6 +557,7 @@ impl Handler {
         let challenge_data = ChallengeData::try_from(packet.authenticated_data().as_slice())
             .expect("Must be the correct challenge size");
         debug!("Sending WHOAREYOU to {}", node_address);
+        self.add_expected_response(node_address.socket_addr);
```

Comment:
I can't right now tell if this is something that we were missing for an already existing `remove_expected_response` call, or if the call is missing after this change

Reply:
I think it was missing before hand and I added it in as I found it.

There was also another bug where we could send a pending request but if it failed we were not reporting back to the service. 

Both of those fixes are in this PR I think.